source/opt/decoration_manager.cpp

```diff
@@ -242,35 +276,26 @@ std::vector<T> DecorationManager::InternalGetDecorationsFor(
 
   const TargetData& target_data = ids_iter->second;
 
+  const auto process_direct_decorations =
+      [include_linkage,
+       &decorations](const std::vector<ir::Instruction*>& direct_decorations) {
+        for (ir::Instruction* inst : direct_decorations) {
+          const bool is_linkage = inst->opcode() == SpvOpDecorate &&
+                                  inst->GetSingleWordInOperand(1u) ==
+                                      SpvDecorationLinkageAttributes;
+          if (include_linkage || !is_linkage) decorations.push_back(inst);
+        }
+      };
+
   // Process |id|'s decorations.
-  decorations.reserve(target_data.direct_decorations.size());
-  for (ir::Instruction* inst : ids_iter->second.direct_decorations) {
-    const bool is_linkage =
-        inst->opcode() == SpvOpDecorate &&
-        inst->GetSingleWordInOperand(1u) == SpvDecorationLinkageAttributes;
-    if (include_linkage || !is_linkage) decorations.push_back(inst);
-  }
+  process_direct_decorations(ids_iter->second.direct_decorations);
```

Comment:
If there are only direct calls to this function, why use a lambda?

Reply:
The lambda is called twice: once on line 291, and once on line 298 inside the loop over groups’ direct decorations.

Reply:
I know it is called twice.  I'm not suggesting it should be inlined.  I'm just wondering why a regular function, where `include_linkage` and `decorations` are passed as parameters,  would not work.

Reply:
Ah sorry, I misunderstood your comment. A regular function could work as well, but I am not sure who else would call it, and since there was only one place needing it, a lambda seemed to be the best. Also, I was thinking I could replace `include_linkage` by a predicate function instead.